text/0000-rand-crypto.md

```diff
@@ -0,0 +1,192 @@
+- Feature Name: rand-crypto
+- Start Date: 2017-08-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Background
+
+This RFC is an offshoot of the [Rand revision RFC], on the subject of:
+
+*   splitting rand into multiple crates
+*   splitting the current `Rng` trait into a crypto-oriented trait and
+    numerical-application-oriented trait
+*   impl of above trait(s) for OS-provided randomness
+*   naming conventions for above things
+
+# Summary
+[summary]: #summary
+
+Create a new crate, `crypto-rand`, with only the following contents:
+
+```rust
+pub enum Error {
+    pub Unspecified,
+}
+
+pub trait Generator {
+    fn try_fill(&self, dest: &mut [u8]) -> Result<(), Error>;
+}
+```
+
+Associated with this, create another crate called `crypto-rng-os` which defines
+a struct `OsGen`, implementing `Generator` via OS functionality (using code from
```

Comment:
If we want to use some prefix for crate names, I think something like `csrng-*` will be better. Although I like suffixes like `os-csrng` a bit more. Also if we use `rng` in the crate name, then it's logical to use `OsRng` instead (plus it will be familiar to users).